src/pal/src/include/pal/stackstring.hpp

```diff
@@ -1,22 +1,30 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information. 
 
-#include "pal/malloc.hpp"
-#include "pal/dbgmsg.h"
+#ifndef __STACKSTRING_H_
+#define __STACKSTRING_H_
 
-SET_DEFAULT_DEBUG_CHANNEL(MISC);
+#include "pal/malloc.hpp"
+#include <type_traits>
 
-template <SIZE_T STACKCOUNT>
+template <SIZE_T STACKCOUNT, class T>
 class StackString
 {
 private:
-    WCHAR m_innerBuffer[STACKCOUNT + 1];
-    WCHAR * m_buffer;
+    T m_innerBuffer[STACKCOUNT + 1];
+    T * m_buffer;
     SIZE_T m_count; // actual allocated count
 
     void NullTerminate()
     {
-        m_buffer[m_count] = W('\0');
+        if (std::is_same<T, CHAR>::value)
```

Comment:
You can just assign integer zero and get rid of this test. That works for both char and WCHAR.